src/plugins/regions.ts

```diff
@@ -279,6 +291,22 @@ class SingleRegion extends EventEmitter<RegionEvents> {
       this.end = options.end ?? (isMarker ? this.start : this.end)
       this.renderPosition()
     }
+    if (this.start !== undefined && this.end !== undefined) {
+      if (options.id !== undefined) {
+        this.id = options.id
+        const isMarker = this.start === this.end
+        if (Array.isArray(this.id)) {
+          this.element.setAttribute('part', `${isMarker ? 'marker' : 'region'} ${this.id.join(' ')}`)
+        } else {
+          this.element.setAttribute('part', `${isMarker ? 'marker' : 'region'} ${this.id}`)
+        }
+      }
```

Comment:
Please move this out of the parent `if`. Start/end props have nothing to do with the id, do they?